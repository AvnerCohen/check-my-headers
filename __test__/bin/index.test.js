const { execSync } = require('child_process')
const path = require('path')

const CliFile = () => path.join(process.cwd(), 'bin/check-my-headers.js')
describe('CLI Behaviour', () => {
  test('Success process: external url', () => {
    const exe = execSync(`node ${CliFile()} http://github.com/ulsiesgascon/check-my-headers`)
    expect(exe.toString()).toMatchSnapshot()
  })

  test('Success process: localhost url', () => {
    const exe = execSync(`node ${CliFile()} localhost:3000`)
    expect(exe.toString()).toMatchSnapshot()
  })

  test('Error process: Missing URL', () => {
    const exe = execSync(`echo ${CliFile()}`)
    expect(exe.toString()).toMatchSnapshot()
  })
})
