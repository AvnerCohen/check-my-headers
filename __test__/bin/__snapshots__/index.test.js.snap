// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI Behaviour Error process: Missing URL 1`] = `
"/Users/ulisesgascon/Desktop/WORKSPACE/my_stuff/check-my-headers/bin/check-my-headers.js
"
`;

exports[`CLI Behaviour Success process: external url 1`] = `
"Welcome to check-my-headers@0.0.1
The analysis has started...
Hello World, this check-my-headers!
"
`;

exports[`CLI Behaviour Success process: localhost url 1`] = `
"Welcome to check-my-headers@0.0.1
The analysis has started...
Hello World, this check-my-headers!
"
`;
